### DISCUSSION TITLE
Calling async C# method from async JavaScript

### DISCUSSION THREAD

**Discussion by Soundman32** [ClearFoundry/ClearScript#598] [Q&A]
Are there any examples on how to call an async C# method from a JS script, and what the methods should look like?

For example, given the following code (ignore _logger, that works fine):

I've enabled TaskPromiseConversion:

`using var engine = new V8ScriptEngine(V8ScriptEngineFlags.EnableTaskPromiseConversion);`

```
javascript:

(async () => {
    _logger.LogInformation("Start");
    await _http.GetAsync("https://cnn.com")
        .then((response) => {
            _logger.LogInformation("R:{response}", response);
        });
    _logger.LogInformation("End");
})();
```
And the C# method is declared like this:

`public Task<TResponse> GetAsync<TResponse>(string url)`

I can see `Start` in the log, but the GetAsync doesn't get called (I have a breakpoint on that line), and `End` isn't in the log.

**Comment by ClearScriptLib** [comment]
Hi @Soundman32,

>`public Task<TResponse> GetAsync<TResponse>(string url)`
>
>I can see `Start` in the log, but the GetAsync doesn't get called (I have a breakpoint on that line),

`GetAsync` requires an explicit type argument for `TResponse`, but your script code doesn't provide one. You need to expose the relevant host type – say, `SomeResponse` – and use the following syntax: `_http.GetAsync(SomeResponse, 'https://cnn.com')`

>and `End` isn't in the log.

It would be helpful to know how you're invoking your script code and awaiting its completion.

Thanks!
